### PR TITLE
Copy slider velocities to strict tracking sliders

### LINKED ISSUE #23471: Slider speed bug when "Strict Tracking" mod enabled
### Type

Game behaviour

### Bug description

When enable "Strict tracking" mod the slider speed is not work properly

### Screenshots or videos

https://github.com/ppy/osu/assets/81455657/43b9b0d7-c84f-49e5-9aec-728d121d3ad2

### Version

2023.511.0

### Logs

[database.log](https://github.com/ppy/osu/files/11451912/database.log)
[input.log](https://github.com/ppy/osu/files/11451913/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/11451914/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/11451915/network.log)
[performance.log](https://github.com/ppy/osu/files/11451916/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/11451917/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/11451918/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/11451920/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/11451921/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/11451922/runtime.log)
[updater.log](https://github.com/ppy/osu/files/11451924/updater.log)

### PR DESCRIPTION
**Pull request by apollo-dw**
Fixes https://github.com/ppy/osu/issues/23471

The `SliderVelocity` property was introduced to `Slider` in https://github.com/ppy/osu/pull/23308, changing the way slider velocities are applied to sliders. To keep things working as expected, this property needs to be also copied over when creating `StrictTrackingSlider`s.

If you want to test this, you can try the [Stalagmite] difficulty on https://osu.ppy.sh/beatmapsets/1006950#osu/2384193 which is currently broken on master.

### CODE REVIEW

**Comment by bdach** [comment]
I will eat some humble pie and admit that the possibility of such a regression [was indirectly pointed out before](https://github.com/ppy/osu/pull/17356#discussion_r830759584). I still think that separating out hitobject classes for mods that change baseline game mechanics is better than having a bunch of flags in the base implementation, though. And I don't think it's going to be especially common to see the slider class receive changes such as https://github.com/ppy/osu/pull/23308 had.

I think we can get this sort of fix in once without test coverage and see how it goes, and reconsider the implementation and the testing if it ever happens again. I'll probably wait for a second approval though and see if the others have a different opinion on whether we want to continue down the derived hitobject path, and whether this needs automated testing.

**Review by bdach — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**
